accounts/tests.py

```diff
@@ -1205,9 +1205,35 @@ def test_change_username_form_admin(self):
         self.assertEqual(OldUsername.objects.filter(user=userA).count(), 2)
 
         # Try to rename for a third time to a valid username. Because we are in admin now, the USERNAME_CHANGE_MAX_TIMES
-        # restriciton does not apply so rename should work correctly
+        # restriction does not apply so rename should work correctly
         post_data.update({'username': u'userANewNewNewName'})
         resp = self.client.post(admin_change_url, data=post_data)
         self.assertRedirects(resp, reverse('admin:auth_user_changelist'))  # Successful edit redirects to users list
         self.assertEqual(OldUsername.objects.filter(username='userANewNewName', user=userA).count(), 1)
         self.assertEqual(OldUsername.objects.filter(user=userA).count(), 3)
+
+
+class UsernameValidatorTests(TestCase):
+    """ Makes sure that username validation works as intended """
+    class TestForm(forms.Form):
+        username = UsernameField()
+
+    def test_valid(self):
+        """ Alphanumerics, _, - and + are ok"""
+        form = self.TestForm(data={'username': 'normal_user.name+'})
+        self.assertTrue(form.is_valid())
+
+    def test_email_like_invalid(self):
+        """ We don't allow @ character """
+        form = self.TestForm(data={'username': 'email@username'})
+        self.assertFalse(form.is_valid())
+
+    def test_short_invalid(self):
+        """ Should be longer than 3 characters """
+        form = self.TestForm(data={'username': 'a'})
+        self.assertFalse(form.is_valid())
+
+    def test_long_invalid(self):
+        """ Should be shorter than 3 characters """
```

Comment:
Instead of 3 it should say 30 here right?

Reply:
updated